lib/dockerregistry/checks.go

```diff
@@ -164,3 +156,73 @@ func (check *ImageRemovalCheck) Compare(
 	}
 	return nil
 }
+
+// MKRealImageSizeCheck returns an instance of ImageSizeCheck which
+// checks that all images to be promoted are under a max size.
+func MKRealImageSizeCheck(
+	maxImageSize int,
+	edges map[PromotionEdge]interface{},
+	digestImageSize DigestImageSize,
+) *ImageSizeCheck {
+	return &ImageSizeCheck{
+		maxImageSize,
+		digestImageSize,
+		edges,
+	}
+}
+
+func (check *ImageSizeCheck) joinImageSizesToString(
+	imageSizes map[string]int,
+) string {
+	imageSizesStr := ""
+	imageNames := make([]string, 0)
+	for k := range imageSizes {
+		imageNames = append(imageNames, k)
+	}
+	const mbToByteShift = 20
+	sort.Strings(imageNames)
+	for i, imageName := range imageNames {
```

Comment:
It would be nicer to consume this data if each image was on its own separate line (assuming that humans will the main consumers of this message).

Reply:
Done